### PR TITLE
Documentation of verifying webhook signatures

### PR DESCRIPTION
**Pull request by z4kn4fein**
### Description

Documentation of verifying the calculated signatures of webhooks sent by ConfigCat.

### Requirement checklist

- [x] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

**Review by laliconfigcat — APPROVED** [review]

File: website/docs/advanced/notifications-webhooks.md [hunk 1]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+
+## Verifying Webhook requests
+To ensure the requests you receive are actually sent by ConfigCat, we highly recommend to verify
+the signature sent in the request headers by comparing it with your own calculated signature.
+
+Each webhook request contains the following headers:
+
+| Header                             | Description                                                                                         |
+| ---------------------------------- | --------------------------------------------------------------------------------------------------- |
+| `X-ConfigCat-Webhook-ID`           | The webhook request's unique identifier.                                                            |
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:66]:
```suggestion
| `X-ConfigCat-Webhook-ID`           | The webhook request's unique identifier. Different in every request. |
```

File: website/docs/advanced/notifications-webhooks.md [hunk 2]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+:::caution
+The signature calculation is sensitive to any changes on the signed content, so it's important to not change the request body before the verification. 
+Otherwise, the calculated signature could be completely different from the value received in the  
+`X-ConfigCat-Webhook-Signature-V1` header.
+:::
+
+### Calculating signature
+ConfigCat uses `HMAC` with `SHA-256` to sign webhooks. You can retrieve the **signing key(s)** required for the signature calculation from the <a href="https://app.configcat.com/product/webhooks" target="_blank">ConfigCat Dashboard's webhook section</a>.
+
+<img src="/docs/assets/whsk.png" className="zoomable bottom-spaced" alt="signing keys" />
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:104]: The censored part should be black or white, only the emphasized part should be red.

File: website/docs/advanced/notifications-webhooks.md [hunk 3]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+| `body`                  | The raw body of the received webhook request. If the webhook doesn't have a request body it can be left out.      |
+
+:::caution
+The signature calculation is sensitive to any changes on the signed content, so it's important to not change the request body before the verification. 
+Otherwise, the calculated signature could be completely different from the value received in the  
+`X-ConfigCat-Webhook-Signature-V1` header.
+:::
+
+### Calculating signature
+ConfigCat uses `HMAC` with `SHA-256` to sign webhooks. You can retrieve the **signing key(s)** required for the signature calculation from the <a href="https://app.configcat.com/product/webhooks" target="_blank">ConfigCat Dashboard's webhook section</a>.
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:102]:
```suggestion
ConfigCat uses `HMAC` with `SHA-256` to sign webhooks. You can retrieve the **signing key(s)** required for the signature calculation from the <a href="https://app.configcat.com/product/webhooks" target="_blank">ConfigCat Dashboard's webhook page</a>.
```

File: website/docs/advanced/notifications-webhooks.md [hunk 4]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+To ensure the requests you receive are actually sent by ConfigCat, we highly recommend to verify
+the signature sent in the request headers by comparing it with your own calculated signature.
+
+Each webhook request contains the following headers:
+
+| Header                             | Description                                                                                         |
+| ---------------------------------- | --------------------------------------------------------------------------------------------------- |
+| `X-ConfigCat-Webhook-ID`           | The webhook request's unique identifier.                                                            |
+| `X-ConfigCat-Webhook-Timestamp`    | The time the webhook was sent in unix timestamp format. (seconds since epoch)                       |
+| `X-ConfigCat-Webhook-Signature-V1` | The list of the `base64` encoded HMAC-SHA-256 signatures. (comma delimited, 1 for each signing key) |
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:68]:
```suggestion
| `X-ConfigCat-Webhook-Signature-V1` | The list of the `base64` encoded HMAC-SHA-256 signatures. (comma delimited, 1 signature for each signing key) |
```

File: website/docs/advanced/notifications-webhooks.md [hunk 5]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+<Tabs>
+<TabItem value="whsk-nodejs" label="Node.js" default>
+
+```js
+const crypto = require("crypto");
+
+// retrieved from ConfigCat Dashboard
+const signingKey = "configcat_whsk_VN3juirnVh5pNvCKd81RYRYchxUX4j3NykbZG2fAy88=" 
+
+// retrieved from X-ConfigCat-Webhook-Signature-V1
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:122]:
```suggestion
// retrieved from the X-ConfigCat-Webhook-Signature-V1 request header
```

File: website/docs/advanced/notifications-webhooks.md [hunk 6]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+```
+:::
+
+<Tabs>
+<TabItem value="whsk-nodejs" label="Node.js" default>
+
+```js
+const crypto = require("crypto");
+
+// retrieved from ConfigCat Dashboard
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:119]:
```suggestion
// retrieved from the ConfigCat Dashboard
```

File: website/docs/advanced/notifications-webhooks.md [hunk 7]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+const signingKey = "configcat_whsk_VN3juirnVh5pNvCKd81RYRYchxUX4j3NykbZG2fAy88=" 
+
+// retrieved from X-ConfigCat-Webhook-Signature-V1
+const receivedSignature = "Ks3cYsu9Lslfo+hVxNC3oQWnsF9e5d73TI5t94D9DRA=" 
+
+// retrieved from X-ConfigCat-Webhook-ID
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:125]:
```suggestion
// retrieved from the X-ConfigCat-Webhook-ID request header
```

File: website/docs/advanced/notifications-webhooks.md [hunk 8]
```diff
@@ -47,7 +51,301 @@ The structure of the JSON array injected into the **##ChangeDetails##** looks li
...
+const requestId = "b616ca659d154a5fb907dd8475792eeb" 
+
+// retrieved from X-ConfigCat-Webhook-Timestamp
```
Comment (sigewuzhere) [inline website/docs/advanced/notifications-webhooks.md:128]:
```suggestion
// retrieved from the X-ConfigCat-Webhook-Timestamp request header
```

File: website/docs/advanced/notifications-webhooks.md (hunk 6 above)
Comment (sigewuzhere) [reply website/docs/advanced/notifications-webhooks.md:119]: please fix everywhere

File: website/docs/advanced/notifications-webhooks.md (hunk 5 above)
Comment (sigewuzhere) [reply website/docs/advanced/notifications-webhooks.md:122]: please fix everywhere

File: website/docs/advanced/notifications-webhooks.md (hunk 7 above)
Comment (sigewuzhere) [reply website/docs/advanced/notifications-webhooks.md:125]: please fix everywhere

File: website/docs/advanced/notifications-webhooks.md (hunk 8 above)
Comment (sigewuzhere) [reply website/docs/advanced/notifications-webhooks.md:128]: please fix everywhere

**Review by sigewuzhere — CHANGES_REQUESTED** [review]

**Review by sigewuzhere — APPROVED** [review]

**Review by laliconfigcat — APPROVED** [review]

**Pull request merged**